internal/pproflogging/pproflogging.go

```diff
@@ -97,10 +103,228 @@ var pprofProfileRates = map[ProfileName]pprofSetRate{
 	},
 }
 
+// MaybeStartProfileBuffers start profile buffers for this process.
+func MaybeStartProfileBuffers(ctx context.Context) {
+	var err error
+
+	pcm, err := LoadProfileConfig(ctx, os.Getenv(EnvVarKopiaDebugPprof))
+	if err != nil {
+		log(ctx).With("error", err).Debug("cannot start configured profile buffers")
+		return
+	}
+
+	if pcm == nil {
+		log(ctx).Debug("no profile buffer configuration to start")
+		return
+	}
+
+	pprofConfigs.mu.Lock()
+	defer pprofConfigs.mu.Unlock()
+
+	pprofConfigs.pcm = pcm
+
+	log(ctx).Debug("no profile buffer configuration to start")
+
+	pprofConfigs.StartProfileBuffersLocked(ctx)
+}
+
+// MaybeRestartProfileBuffers used by SIGQUIT signal handlers to output PPROF data without exiting.
+func MaybeRestartProfileBuffers(ctx context.Context) {
+	pprofConfigs.mu.Lock()
+	defer pprofConfigs.mu.Unlock()
+
+	if pprofConfigs == nil || len(pprofConfigs.pcm) == 0 {
```

Comment:
This line would not be reached if `pprofConfigs == nil`, since it would have panicked above.